File: backend/src/main/java/de/neuefische/backend/dto/TripDto.java
package de.neuefische.backend.dto;

import de.neuefische.backend.model.*;
import lombok.*;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import java.time.LocalDate;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class TripDto {
    @NotBlank(message = "Title is mandatory")
    private String title;

    private String destinationCountry;

    @NotNull(message = "Traveller amount is mandatory")
    private double travellerAmount;

    @NotNull(message = "Date of Departure is mandatory")
    private LocalDate dateOfDeparture;

    @NotNull(message = "Date of Returning is mandatory")
    private LocalDate dateOfReturning;

    @NotNull(message = "Personal CO2 Budget is mandatory" )
    private double personalBudget;

    private List<Transportation> transportations;
    private List <Accommodation> accommodations;
    private List <Food> foods;
    private List <Shopping> shoppings;
    private List <Activity> activities;
}

File: frontend/src/model/TripDto.ts
import {Accommodation, Activity, Food, Shopping, Transportation} from "./Trip";

export type TripDto = {

    title: string;
    destiniationCountry: string;
    travellerAmount: number;
    dateOfDeparture: string;
    dateOfReturning: string;
    personalBudget: number;
    transportations: Transportation[];
    accommodations: Accommodation[];
    foods: Food[];
    shoppings: Shopping[];
    activities: Activity[]
}